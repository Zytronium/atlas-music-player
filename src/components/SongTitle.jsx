export default function SongTitle(props) {
  return (
    <div className="flex flex-col">
      <h2 className={props.size === 'small' ? 'text-md' : 'text-2xl font-bold'}>{props.title}</h2>
      <p className={`text-gray-500 ${props.size === 'small' ? 'text-md' : ''}`}>{props.artist}</p>
    </div>
  );
}
