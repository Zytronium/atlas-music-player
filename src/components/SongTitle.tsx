import { JSX } from "react";

type Props = {
  title: string;
  artist: string;
  size?: "small" | "large";
};

export default function SongTitle(props: Props): JSX.Element {
  return (
    <div className="flex flex-col">
      <h2
        className={`${props.size === "small" ? "text-sm font-[500]" : "text-line mb-2 text-2xl/8 font-bold"} text-left`}
      >
        {props.title}
      </h2>
      <p
        className={`dark:text-dark-muted text-muted ${props.size === "small" ? "text-sm font-[500]" : "mb-4"} text-left`}
      >
        {props.artist}
      </p>
    </div>
  );
}
