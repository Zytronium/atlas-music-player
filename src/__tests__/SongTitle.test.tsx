import { render } from "@testing-library/react";
import { expect, test } from "vitest";
import SongTitle from "../components/SongTitle";

test("SongTitle - default/large styling (size omitted)", () => {
  const { container } = render(
    <SongTitle title="The Atlas" artist="16 / 16 / 16" />,
  );
  expect(container).toMatchSnapshot();
});

test("SongTitle - small size styling", () => {
  const { container } = render(
    <SongTitle title="Short Track" artist="Some Artist" size="small" />,
  );
  expect(container).toMatchSnapshot();
});

test("SongTitle - long title and artist (wrap/overflow cases)", () => {
  const longTitle =
    "This is an unusually long song title that is intended to test wrapping and layout behavior in the component that should certainly overflow unless your monitor or zoom level is simply absurdly, astronomically, nearly incomprehensible large as heck like seriously who hsa a screen this large I know I don't except maybe the Atlas School Main Stage display can display it all if you adjust the zoom level";
  const longArtist =
    "A Very Long Artist Name That Should Overflow Or Wrap In Narrow Layouts Especially On Smaller Screens Like Mobile Phones";
  const { container } = render(
    <SongTitle title={longTitle} artist={longArtist} />,
  );
  expect(container).toMatchSnapshot();
});

test("SongTitle - empty title and artist (edge case)", () => {
  const { container } = render(<SongTitle title="" artist="" />);
  expect(container).toMatchSnapshot();
});

test("SongTitle - unicode, emojis, and special characters", () => {
  const { container } = render(
    <SongTitle
      title="🌌 星の旅 — Stårnë"
      artist="Æther & Ωmega 🎵"
      size="small"
    />,
  );
  expect(container).toMatchSnapshot();
});
