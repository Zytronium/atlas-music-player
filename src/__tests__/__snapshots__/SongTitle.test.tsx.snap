// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle - default/large styling (size omitted) 1`] = `
<div>
  <div
    class="flex flex-col"
  >
    <h2
      class="text-line mb-2 text-2xl/8 font-bold text-left"
    >
      The Atlas
    </h2>
    <p
      class="dark:text-dark-muted text-muted mb-4 text-left"
    >
      16 / 16 / 16
    </p>
  </div>
</div>
`;

exports[`SongTitle - empty title and artist (edge case) 1`] = `
<div>
  <div
    class="flex flex-col"
  >
    <h2
      class="text-line mb-2 text-2xl/8 font-bold text-left"
    />
    <p
      class="dark:text-dark-muted text-muted mb-4 text-left"
    />
  </div>
</div>
`;

exports[`SongTitle - long title and artist (wrap/overflow cases) 1`] = `
<div>
  <div
    class="flex flex-col"
  >
    <h2
      class="text-line mb-2 text-2xl/8 font-bold text-left"
    >
      This is an unusually long song title that is intended to test wrapping and layout behavior in the component that should certainly overflow unless your monitor or zoom level is simply absurdly, astronomically, nearly incomprehensible large as heck like seriously who hsa a screen this large I know I don't except maybe the Atlas School Main Stage display can display it all if you adjust the zoom level
    </h2>
    <p
      class="dark:text-dark-muted text-muted mb-4 text-left"
    >
      A Very Long Artist Name That Should Overflow Or Wrap In Narrow Layouts Especially On Smaller Screens Like Mobile Phones
    </p>
  </div>
</div>
`;

exports[`SongTitle - small size styling 1`] = `
<div>
  <div
    class="flex flex-col"
  >
    <h2
      class="text-sm font-[500] text-left"
    >
      Short Track
    </h2>
    <p
      class="dark:text-dark-muted text-muted text-sm font-[500] text-left"
    >
      Some Artist
    </p>
  </div>
</div>
`;

exports[`SongTitle - unicode, emojis, and special characters 1`] = `
<div>
  <div
    class="flex flex-col"
  >
    <h2
      class="text-sm font-[500] text-left"
    >
      🌌 星の旅 — Stårnë
    </h2>
    <p
      class="dark:text-dark-muted text-muted text-sm font-[500] text-left"
    >
      Æther & Ωmega 🎵
    </p>
  </div>
</div>
`;
